### ISSUE TITLE
MOB support for PC128 / MO5

### ISSUE THREAD

**Issue by spotlessmind1975** [spotlessmind1975/ugbasic#177]
Implementation for MOBs is required on Thomson MO5 and Olivetti Prodest PC128.

**Comment by dinoflorenzi** [comment]
You have implemented GET IMAGE AND PUT IMAGE. Can you merge them to implement MOB?

**Comment by spotlessmind1975** [comment]
Hi @dinoflorenzi **it would be great if it were possible!**

It's actually not that simple.

`MOB` was created to simulate sprites in those environments where they are not natively available. In practice it means giving the possibility to position an image in any arbitrary position (pixel alignment), with the management of overlaps and all the rest. If you move a `MOB` you shouldn't worry about anything other than that.

`PUT IMAGE` / `GET IMAGE`, on the other hand, are primitives that work with byte, not pixel, alignment. This means that it is not possible to position them arbitrarily (horizontally) but by multiples of `2`, `4` or even `8` pixels (depending on the resolution of the screen).

There is an issue that I have opened to overcome this limit (#208) but if you see the analysis you will discover that I have encountered a series of non-trivial problems: from the (pre)calculation of the shifts to the need to calculate the masking.

Also, and it's a very serious problem, as long as it's a single image that's okay but when it's an animation, things get dramatic.

So, I am sorry but I think it follows that it is not possible to use them directly to simulate MOBs.
